transcom/mymove
# B 21041 Prime can update existing second/third addresses

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21041)
## [INT PR](https://github.com/transcom/mymove/pull/13702)

> [!NOTE]
> I didn't follow best practices and update my main branch and merge that into my integration branch for the final commit, so please pay close attention to changes to mto_shipment_address.go

## Summary

Part of the effort to allow prime to view/update/add third addresses (also includes the work to add second addresses). This work allows prime to update addresses for existing shipments that have second/third addresses. In planning my approach, I thought it would be quicker to refactor the existing Prime UI update address component, and make it better for those of us who use Prime SIM, so I took the liberty of doing that. Now, when you click update on an address, it will take you an update address page for that specific address.

### How to test
Setup: if [21054](https://github.com/transcom/mymove/pull/13692) has not been merged you will either need a move with shipments of all types, with second/third pickup/delivery addresses

1. Access the move described above as prime
2. On the move details screen, scroll to a shipment with addresses and click 'Edit' next to an address
3. At the new screen, confirm that the details are all accurate
4. Update the data and click save and confirm data has properly saved
5. Try with all combinations of shipments/addresses
NOTE: NTS/NTSR will not update if a primary pickup or destination address is missing, its an existing bug
NOTE: HHG Primary destination addresses will not update, as they have a separate endpoint to use
NOTE: Updating PPM second/third addresses is being handled in [21054](https://github.com/transcom/mymove/pull/13692)
